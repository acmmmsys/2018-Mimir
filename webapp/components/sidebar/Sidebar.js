import React, { Component } from 'react'
import { Link } from 'react-router-dom'

import SidebarSection from 'components/sidebar/SidebarSection';
import SidebarMenu from 'components/sidebar/SidebarMenu';
import SidebarMenuItem from 'components/sidebar/SidebarMenuItem';

import 'style/sidebar/SideBar.scss';

class Sidebar extends Component {
    render() {
        return (
            <nav className='sidebar ui-sidebar border-right-dark'>
                <header className='ui-sidebar-header'>
                    Gastroenterology
                </header>
                <SidebarSection title='Report Generation'>
                    <SidebarMenu>
                        <SidebarMenuItem>
                            <Link to='/image'>Image Analysis</Link>
                        </SidebarMenuItem>
                        <SidebarMenuItem>
                            <Link to='/report'>Reports</Link>
                        </SidebarMenuItem>
                    </SidebarMenu>
                </SidebarSection>
            </nav>
        )
    }
}

export default Sidebar;