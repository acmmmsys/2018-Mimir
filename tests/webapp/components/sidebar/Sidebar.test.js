import React from 'react';
import Sidebar from '../../../../webapp/components/sidebar/Sidebar'
import renderer from 'react-test-renderer';

let wrapper;

test('Sidebar does render', () => {
    wrapper = shallow(
        <Sidebar>

        </Sidebar>
    );
    expect(wrapper).toMatchSnapshot();
});
