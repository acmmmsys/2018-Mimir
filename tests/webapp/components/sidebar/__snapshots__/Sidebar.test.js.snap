// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar does render 1`] = `
<nav
  className="sidebar"
>
  <header>
    Gastroenterology
  </header>
  <SidebarSection
    title="Report Generation"
  >
    <SidebarMenu>
      <SidebarMenuItem>
        <Link
          replace={false}
          to="/image"
        >
          Image Analysis
        </Link>
      </SidebarMenuItem>
      <SidebarMenuItem>
        <Link
          replace={false}
          to="/report"
        >
          Reports
        </Link>
      </SidebarMenuItem>
    </SidebarMenu>
  </SidebarSection>
</nav>
`;
